### PR TITLE
Fibbonacci sum solution added

### LINKED ISSUE #388: [Feature Request]: i want to add solution of fibbonacci sum problem
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

i want to add solution of fibbonacci sum problem

### Use Case

user can easily see solution if they did not got solution after very try.

### Benefits

user can easily see solution if they did not got solution after very try.

### Add ScreenShots

Screenshot 2024-06-03 141600

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Vipullakum007**
## Related Issue
i want to add solution of fibbonacci sum problem #388

fixes #388 
[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

i added solution to fibonacci sum problem

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

https://github.com/CodeHarborHub/codeharborhub/assets/127659443/d829e0d4-677c-4b82-98c8-ca548bea9a9a

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+By leveraging the properties of Fibonacci numbers and matrix multiplication, we can achieve optimal performance for this problem, making it suitable for large input sizes as specified in the constraints.
+
+## References
+
+- **GeeksforGeeks Problem:** [Geeks for Geeks Problem](https://www.geeksforgeeks.org/problems/fibonacci-sum/0)
+- **Solution Link:** [Fibonacci Sum on Geeks for Geeks](https://www.geeksforgeeks.org/problems/fibonacci-sum/0)
+- **Authors GeeksforGeeks Profile:** [Vipul](https://www.geeksforgeeks.org/user/lakumvipwjge/)
+
+```
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md:271]: remove line 271 to line 273

File: dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  - Dynamic Programming
+  - Mathematics
+  - Algorithms
+---
+
+This tutorial contains a complete walk-through of the Fibonacci Sum problem from the Geeks for Geeks website. It features the implementation of the solution code in three programming languages: Python, C++, and Java.
+
+## Problem Description
+
+Given a positive number N, find the value of f0 + f1 + f2 + ... + fN where fi indicates the ith Fibonacci number. Note that f0 = 0, f1 = 1, f2 = 1, f3 = 2, f4 = 3, f5 = 5, and so on. Since the answer can be very large, return the result modulo 1000000007.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md:17]: **Basic Practice:**

```md
f0 + f1 + f2 + ... + fN

f0 = 0, f1 = 1, f2 = 1, f3 = 2, f4 = 3, f5 = 5,

1000000007
```

**Good Practice:**

```md
$f0 + f1 + f2 + ... + fN$

$f0 = 0, f1 = 1, f2 = 1, f3 = 2, f4 = 3, f5 = 5,$

$1000000007$
```

File: dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+Input:
+N = 3
+Output:
+4
+Explanation:
+0 + 1 + 1 + 2 = 4
+```
+
+```
+**Example 2:**
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md:33]: write `**Example 2:**` out of <code>```</code>

File: dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+This tutorial contains a complete walk-through of the Fibonacci Sum problem from the Geeks for Geeks website. It features the implementation of the solution code in three programming languages: Python, C++, and Java.
+
+## Problem Description
+
+Given a positive number N, find the value of f0 + f1 + f2 + ... + fN where fi indicates the ith Fibonacci number. Note that f0 = 0, f1 = 1, f2 = 1, f3 = 2, f4 = 3, f5 = 5, and so on. Since the answer can be very large, return the result modulo 1000000007.
+
+## Examples
+
+```
+**Example 1:**
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md:22]: write `**Example 1:**` out of <code>```</code>

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update contents based on suggestions

**Comment by Vipullakum007** [comment]
hey @Ajay-Dhangar , i have updated as per your suggestion.

File: dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md (hunk 2 above)
Comment (Vipullakum007) [reply dsa-solutions/gfg-solutions/0004-fibbonacci-sum.md]: hey @Ajay-Dhangar , you very kind person thx for suggestion.

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**